Sources/SpeziHealthKit/Bulk Export/BulkHealthExporter+Session.swift

```diff
@@ -0,0 +1,378 @@
+                } else {
+                    let start = DateFormatter.localizedString(from: timeRange.lowerBound, dateStyle: .short, timeStyle: .none)
+                    let end = DateFormatter.localizedString(from: timeRange.upperBound.advanced(by: -1), dateStyle: .short, timeStyle: .none)
+                    desc += "(\(start) – \(end))"
+                }
+                return desc
+            }
+            
+            init(sampleType: WrappedSampleType, timeRange: Range<Date>) {
+                self.sampleType = sampleType
+                self.timeRange = timeRange
+                self.shouldSkipUntilNextLaunch = false
+            }
+        }
+        
+        let sessionId: String
+        let exportEndDate: Date
+        var pendingBatches: [ExportBatch]
+        var completedSampleTypes: [WrappedSampleType]
+        var numCompletedExportBatches: Int
+        
+        init(sessionId: String, exportEndDate: Date, sampleTypes: Set<WrappedSampleType>, using healthKit: HealthKit) async {
+            self.sessionId = sessionId
+            self.exportEndDate = exportEndDate
+            self.completedSampleTypes = []
+            self.pendingBatches = []
+            self.numCompletedExportBatches = 0
+            for sampleType in sampleTypes {
+                await add(sampleType: sampleType, healthKit: healthKit)
+            }
+        }
+        
+        mutating func add(sampleType: WrappedSampleType, healthKit: HealthKit) async {
+            guard !(completedSampleTypes.contains(sampleType) || pendingBatches.contains { $0.sampleType == sampleType }) else {
+                // we've either already marked the sample type as completed, or have it already scheduled
+                // --> nothing to be done
+                return
+            }
+            let cal = Calendar(identifier: .gregorian)
+            let endDate = self.exportEndDate
+            let startDate: Date = (try? await sampleType.oldestSampleDate(in: healthKit)) ?? {
+                cal.date(from: .init(year: 2014, month: 6, day: 2))! // swiftlint:disable:this force_unwrapping
+            }()
+            let yearRanges = sequence(first: cal.rangeOfYear(for: startDate)) {
+                $0.contains(endDate) || $0.lowerBound >= endDate ? nil : cal.rangeOfYear(for: cal.startOfNextYear(for: $0.lowerBound))
+            }
+            pendingBatches.append(contentsOf: yearRanges.map { year in
+                ExportBatch(sampleType: sampleType, timeRange: year)
+            })
+        }
+    }
+    
+    
+    /// A long-running backgrund exporting task that fetches and processes HealthKit data.
+    ///
+    /// ## Topics
+    /// ### Session Lifecycle Management
+    /// - ``start()``
+    /// - ``pause()``
+    /// ### Session State
+    /// - ``isRunning``
+    /// - ``state``
+    /// - ``progress``
+    @Observable
+    public final class ExportSession<Processor: BatchProcessor>: Sendable, ExportSessionProtocol {
```

Comment:
the type is currently called an `ExportSession`. previously it was just called a `Session`. i'm not really happy with either of these names, we might want to call it an `ExportTask` instead?